src/omotes_simulator_core/entities/esdl_object.py

```diff
@@ -70,21 +68,17 @@ def get_connected_assets(self, asset_id: str, port: Port) -> List[Tuple[str, Por
         with the type of port they are connected to.
 
         :param str id: id of the asset for which we want to know the connected assets
-        :param Port port: port for which the connected assets need to be returned.
-        :return: List of list which the id of the connected assets and the connected port.
+        :param str port: port for which the connected assets need to be returned.
+        :return: List of list which the id of the connected assets and the connected port id.
         """
-        # TODO 1. Add support for components with multiple in and outports, like heat exchanger
         # TODO 2. What if it is connected to a joint?
         connected_assets = []
         esdl_asset = self.energy_system_handler.get_by_id(asset_id)
 
-        type_port = OutPort if port == Port.Out else InPort
         connected_port_ids = []
         for esdl_port in esdl_asset.port:
-            if isinstance(esdl_port, type_port):
+            if esdl_port.id == port_id:
                 connected_port_ids = esdl_port.connectedTo
```

Comment:
Moet dit niet een append statement zijn? Of is de uitvoer van esdl_port.connectedTo een list? Als dat zo is zou ik verwachten dat de 'connected_port_ids' wordt gereset binnen de loop over esdl_asset.port

Reply:
je break uit de loop zodra je port gevonden hebt waarvoor je de verbonden port ids wilt hebben. Dus zo klopt het. Je creeert eerst een lege lijst anders gaat het fout in een paar regels verder.